modules/service/src/main/scala/lucuma/odb/graphql/mapping/GmosNorthLongSlitMapping.scala

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+
+package mapping
+
+import cats.syntax.all.*
+import edu.gemini.grackle.Mapping
+import edu.gemini.grackle.Path
+import edu.gemini.grackle.Predicate
+import edu.gemini.grackle.Predicate.*
+import edu.gemini.grackle.Query
+import edu.gemini.grackle.Query.*
+import edu.gemini.grackle.Result
+import edu.gemini.grackle.TypeRef
+import edu.gemini.grackle.skunk.SkunkMapping
+import eu.timepit.refined.types.numeric.PosDouble
+import lucuma.core.enums.GmosAmpGain
+import lucuma.core.enums.GmosAmpReadMode
+import lucuma.core.enums.GmosNorthFpu
+import lucuma.core.enums.GmosRoi
+import lucuma.core.enums.GmosXBinning
+import lucuma.core.enums.GmosYBinning
+import lucuma.core.enums.ImageQuality
+import lucuma.core.enums.StellarLibrarySpectrum
+import lucuma.core.model.Observation
+import lucuma.core.model.SourceProfile
+import lucuma.core.model.SpectralDefinition
+import lucuma.core.model.UnnormalizedSED
+import lucuma.odb.graphql.binding.*
+import lucuma.odb.graphql.input.*
+import lucuma.odb.graphql.predicate.Predicates
+import lucuma.odb.graphql.table.*
+
+import scala.collection.immutable.SortedMap
+
+trait GmosNorthLongSlitMapping[F[_]]
+  extends GmosNorthLongSlitTable[F] { this: SkunkMapping[F] =>
+
+  lazy val GmosNorthLongSlitMapping: ObjectMapping =
+    ObjectMapping(
+      tpe = GmosNorthLongSlitType,
+      fieldMappings = List(
+        SqlField("observationId", GmosNorthLongSlitTable.ObservationId, key = true, hidden = true),
+
+        SqlField("grating", GmosNorthLongSlitTable.Grating),
+        SqlField("filter",  GmosNorthLongSlitTable.Filter),
+        SqlField("fpu",     GmosNorthLongSlitTable.Fpu),
+        SqlObject("centralWavelength"),
+    
+        // xBin
+        CursorField(
+          "xBin",
+          cursor =>
+            (cursor.field("explicitXBin", None).flatMap(_.as[Option[GmosXBinning]]),
+             cursor.field("defaultXBin",  None).flatMap(_.as[GmosXBinning])
+            ).parMapN(_.getOrElse(_))
+          ,
+          List("defaultXBin", "explicitXBin")
+        ),
+
+        // TODO: a function of FPU, target SourceProfile, and ImageQuality
+        FieldRef[GmosNorthFpu]("fpu")
+          .as(
+            "defaultXBin",
+            fpu => GmosLongSlitMath.xbinNorth(
+              fpu,
+              /*placeholder*/ SourceProfile.Point(SpectralDefinition.BandNormalized(UnnormalizedSED.StellarLibrary(StellarLibrarySpectrum.O5V), SortedMap.empty)),
+              /*placeholder*/ ImageQuality.TwoPointZero,  // so the FPU size is the effective slit width for now
+              PosDouble.unsafeFrom(2.0)
```

Comment:
Now updated with placeholder `SourceProfile` and `ImageQuality` values.  It's a shame that we need to pass the entire `SourceProfile`.  The `SpectralDefinition` is irrelevant for the purpose of calculating the x-binning.